pkg/util/oom/oom_linux.go

```diff
@@ -0,0 +1,108 @@
+// +build cgo,linux
+
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oom
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"strconv"
+
+	"github.com/docker/libcontainer/cgroups/fs"
+	"github.com/docker/libcontainer/configs"
+	"github.com/golang/glog"
+)
+
+func NewOomAdjuster() *OomAdjuster {
+	oomAdjuster := &OomAdjuster{
+		pidLister:        getPids,
+		ApplyOomScoreAdj: applyOomScoreAdj,
+	}
+	oomAdjuster.ApplyOomScoreAdjContainer = oomAdjuster.applyOomScoreAdjContainer
+	return oomAdjuster
+}
+
+func getPids(cgroupName string) ([]int, error) {
+	fsManager := fs.Manager{
+		Cgroups: &configs.Cgroup{
+			Name: cgroupName,
+		},
+	}
+	return fsManager.GetPids()
+}
+
+// Writes 'value' to /proc/<pid>/oom_score_adj. PID = 0 means self
+func applyOomScoreAdj(pid int, oomScoreAdj int) error {
+	if pid < 0 {
+		return fmt.Errorf("invalid PID %d specified for oom_score_adj", pid)
+	}
+
+	var pidStr string
+	if pid == 0 {
+		pidStr = "self"
+	} else {
+		pidStr = strconv.Itoa(pid)
+	}
+
+	oomScoreAdjPath := path.Join("/proc", pidStr, "oom_score_adj")
+	oom_value, err := ioutil.ReadFile(oomScoreAdjPath)
+	if err != nil {
```

Comment:
I'm going to retry once on any failure. Also, fixed another bug in the applyOomScoreAdj code that I moved from util - I should have read the code in more detail. Thanks for the great catch!